### PR TITLE
Extend code generation with auto formatting

### PR DESCRIPTION
**Pull request by nagyandras95**
The CodeFormatter contained by CDT is used for auto formatting with default options. It is involved to add a new feature dependency for the target defnitions and feature.xml file.

### CODE REVIEW

File: dev/features/hu.elte.txtuml.feature/feature.xml [hunk 1]
```diff
@@ -116,6 +116,7 @@ This Agreement is governed by the laws of the State of New York and the intellec
       <import feature="org.eclipse.xtext.xbase.lib" version="2.8.4.v201508050135" match="perfect"/>
       <import feature="org.eclipse.xtext.ui" version="2.8.4.v201508050135" match="perfect"/>
       <import feature="org.eclipse.xtext.xtext.ui" version="2.8.4.v201508050135" match="perfect"/>
+      <import feature="org.eclipse.cdt" version="8.8.0.201509131935"/>
```
Comment (djnemeth) [inline dev/features/hu.elte.txtuml.feature/feature.xml]: @nagyandras95, have you considered setting a stricter version match rule? See cf962cd.

**Comment by nagyandras95** [comment]
@ELTE-Soft/all 
Somebody review, please!

**Comment by djnemeth** [comment]
@nagyandras95, are you sure the `perfect` match rule is necessary for the introduced dependency? See #431.

**Comment by djnemeth** [comment]
As I can see, only the code generated from model classes is formatted. I suggest to extend auto formatting to other model elements (e.g. associations) before merging this pull request.

File: dev/plugins/hu.elte.txtuml.project/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -10,6 +10,7 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.ui.ide,
  org.eclipse.core.runtime,
  org.eclipse.jdt.core,
+ org.eclipse.cdt.core,
```
Comment (djnemeth) [inline dev/plugins/hu.elte.txtuml.project/META-INF/MANIFEST.MF]: Why is CDT listed here as a plugin dependency? It seems nothing changes without it.

**Comment by djnemeth** [comment]
I have tested this pull request by installing txtUML from the generated update site in a fresh Eclipse, and by generating C++ code from all demo models which contain deployment configurations.

Looks good to me! 👍

**Pull request merged**